zipline/_protocol.pyx

```diff
@@ -49,12 +45,18 @@ cdef class BarData:
         try:
             view = self._views[asset]
         except KeyError:
-            view = self._views[asset] = \
-                SidView(asset, self.data_portal,
-                        self.simulator, self.data_frequency)
+            view = self._views[asset] = self._create_sid_view(asset)
 
         return view
 
+    def _create_sid_view(self, asset):
```

Comment:
Maybe add a comment how about this a 'hook' into which a custom sid view can be implemented?